client/extensions/woocommerce/app/settings/payments/index.js

```diff
@@ -36,14 +39,44 @@ class SettingsPayments extends Component {
 		className: PropTypes.string,
 	};
 
+	componentDidMount = () => {
+		const { site } = this.props;
+
+		if ( site && site.ID ) {
+			this.props.fetchSetupChoices( site.ID );
+		}
+	}
+
+	componentWillReceiveProps = ( newProps ) => {
+		const { site } = this.props;
+
+		const newSiteId = newProps.site ? newProps.site.ID : null;
+		const oldSiteId = site ? site.ID : null;
+
+		if ( oldSiteId !== newSiteId ) {
+			this.props.fetchSetupChoices( newSiteId );
+		}
+	}
+
 	onSave = () => {
-		const { translate } = this.props;
+		const { translate, site, finishedInitialSetup } = this.props;
 
-		const successAction = successNotice(
+		let successAction = successNotice(
 			translate( 'Payment settings saved.' ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **27** times:
`translate( 'Payment Settings' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).